packages/alpinejs/src/directives/x-for.js

```diff
@@ -20,7 +20,23 @@ directive('for', (el, { expression }, { effect, cleanup }) => {
     el._x_prevKeys = []
     el._x_lookup = {}
 
-    effect(() => loop(el, iteratorNames, evaluateItems, evaluateKey))
+    if (modifiers.includes('hydrate')) {
+        const intKey = modifiers[modifiers.indexOf('hydrate') + 1] === 'int'
+        let curEl = el;
+        while (curEl = curEl.nextElementSibling) {
+            const key = curEl.getAttribute(':key')
+                curEl.getAttribute(':key')
+            if (key === null) {
+                break
+            }
+
+            const castKey = intKey ? parseInt(key) : key
+            el._x_lookup[castKey] = curEl
+            el._x_prevKeys.push(castKey)
```

Comment:
ah, okay, now I see where you meant you had issues with there being int keys.

Reply:
Yes, if the keys from the `:key` attribute weren't the same type as the keys in the `x-data`, it would treat them as different items.

Reply:
Yes, since mine just used the calculated keys and not the ones directly from the dom, it avoided this...
hmm...